packages/circuit-ui/components/DateInput/components/Dialog.tsx

```diff
@@ -74,6 +89,7 @@ export const Dialog = forwardRef<HTMLDialogElement, DialogProps>(
       }
 
       if (open) {
```

Comment:
What about the code branch on L100, when the `dialog` is closed by changing the `open` attribute to false?